### ISSUE TITLE
Obtaining duals of convex problem with second order cone constraints not possible using Gurobi optimizer

### ISSUE THREAD

**Issue by junglegobs** [JuliaOpt/JuMP.jl#1994]
I have a convex problem (which therefore should have duals) yet these don't seem to work. I can solve the problem just fine, but if I try to obtain the duals of a constraint I either get `NaN` or `AssertionError: dual >= 0.0`. Minimum working example below, I'm guessing it might be specific to Gurobi?

```
using JuMP
using Gurobi

m = Model(with_optimizer(Gurobi.Optimizer))

@variable(m, x >= 0)
@variable(m, y >= 0)
@variable(m, z >= 0)

@constraint(m, con1, x + y == 2)
@constraint(m, con2, x + y + z >= 0)
@constraint(m, con3, x^2 >= y^2 + z^2)

optimize!(m)

dual(con1)
dual(con2)
dual(con3)
```

**Comment by odow** [comment]
Yes, this is specific to Gurobi. I'm currently re-writing the wrapper (https://github.com/JuliaOpt/Gurobi.jl/pull/216) and something weird is going on. I'll investigate.

I'm going to transfer this issue to Gurobi.jl.
